### PR TITLE
adds avahi base service to init function of mdns controller

### PR DESCRIPTION
**Pull request by ProofOfKeags**

### CODE REVIEW

File: appmgr/src/net/mdns.rs [hunk 1]
```diff
@@ -93,6 +93,19 @@ impl MdnsControllerInner {
...
+                avahi_entry_group_add_service(
+                    group,
+                    avahi_sys::AVAHI_IF_UNSPEC,
+                    avahi_sys::AVAHI_PROTO_UNSPEC,
+                    avahi_sys::AvahiPublishFlags_AVAHI_PUBLISH_USE_MULTICAST,
+                    hostname_raw,
+                    http_tcp_cstr.as_ptr(),
+                    std::ptr::null(),
+                    hostname_raw,
+                    5959,
```
Comment (ProofOfKeags) [inline appmgr/src/net/mdns.rs:107]:
```suggestion
                    443,
```

File: appmgr/src/net/mdns.rs [hunk 2]
```diff
@@ -93,6 +93,19 @@ impl MdnsControllerInner {
...
+                    .expect("Could not cast _http._tcp to c string");
+                avahi_entry_group_add_service(
+                    group,
+                    avahi_sys::AVAHI_IF_UNSPEC,
+                    avahi_sys::AVAHI_PROTO_UNSPEC,
+                    avahi_sys::AvahiPublishFlags_AVAHI_PUBLISH_USE_MULTICAST,
+                    hostname_raw,
+                    http_tcp_cstr.as_ptr(),
+                    std::ptr::null(),
+                    hostname_raw,
```
Comment (ProofOfKeags) [inline appmgr/src/net/mdns.rs:106]:
```suggestion
                    std::ptr::null(),
```

**Comment by ProofOfKeags** [comment]
Add hostnamectl change to startup sequence

**Comment by ProofOfKeags** [comment]
ready for re-review

File: appmgr/src/bin/embassy-init.rs [hunk 3]
```diff
@@ -4,6 +4,9 @@ async fn inner_main() -> Result<(), Error> {
     // os sync
     embassy::volume::disk::mount("/dev/sda", "/mnt/embassy-os-crypt").await?;
 
+    // hostname-set
+    embassy::hostname::ensure_correct_hostname().await?;
```
Comment (dr-bonez) [inline appmgr/src/bin/embassy-init.rs:8]:
```suggestion
    embassy::hostname::sync_hostname().await?;
```

File: appmgr/src/hostname.rs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        .await?;
+    Ok(())
+}
+
+pub async fn get_product_key() -> Result<String, Error> {
+    let out = tokio::fs::read_to_string("/boot/product_key.txt").await?;
+    Ok(out.trim().to_owned())
+}
+
+pub async fn ensure_correct_hostname() -> Result<(), Error> {
```
Comment (dr-bonez) [inline appmgr/src/hostname.rs:29]:
```suggestion
pub async fn sync_hostname() -> Result<(), Error> {
```

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**